### ISSUE TITLE
关于作者给出代码中Bert的最大输入长度与论文中不符

### ISSUE THREAD

**Issue by gaozhiguang** [jiacheng-xu/DiscoBERT#11]
作者代码：
if bert_max_length > 512: 
            first_half = self.embedder.bert_model.embeddings.position_embeddings.weight#前一半512个字符
            # ts = torch.zeros_like(first_half, dtype=torch.float32)
            # second_half = ts.new_tensor(first_half, requires_grad=True)

            second_half = torch.zeros_like(first_half, dtype=torch.float32, requires_grad=True)

            # second_half = torch.empty(first_half.size(), dtype=torch.float32,requires_grad=True)
            # torch.nn.init.normal_(second_half, mean=0.0, std=1.0)
            out = torch.cat([first_half, second_half], dim=0)#那么这里的最大position应该是512*2了吧，与文中不符啊
            self.embedder.bert_model.embeddings.position_embeddings.weight = torch.nn.Parameter(out)
            self.embedder.bert_model.embeddings.position_embeddings.num_embeddings = 512 * 2
            self.embedder.max_pieces = 512 * 2
这里很明显最大长度是1024啊，作者论文中说的是768， @jiacheng-xu

**Comment by jiacheng-xu** [comment]
你好，我如果没记错的话我尝试了768和1024，在CNNDM上最后效果一样（ROUGE-1差0.01之类的）。你可以尝试一下。

**Comment by jiacheng-xu** [comment]
[Max len during data preparation](https://github.com/jiacheng-xu/DiscoBERT/blob/814c741e2a049de3afc489835e0df3ccf9fb4fe9/data_preparation/data_structure.py#L20) is set to 768.
